Code/Framework/AzCore/AzCore/Math/RayIntersection.h

```diff
@@ -0,0 +1,21 @@
+
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "AzCore/Math/Vector2.h"
+namespace AZ
+{
+    class Ray;
+    class Vector3;
+
+    namespace RayIntersection
```

Comment:
The vector2 include here can be converted to a forward declaration.
It gets included in the `RayIntersection.inl`, so an include isn't needed in the header.
```suggestion
namespace AZ
{
    class Ray;
    class Vector2
    class Vector3;

    namespace RayIntersection
```